src/Features/Core/Portable/FileHeaders/AbstractFileHeaderCodeFixProvider.cs

```diff
@@ -0,0 +1,237 @@
+            var newLineTrivia = EndOfLine(newLineText);
+            var newTrivia = CreateNewHeader(FileHeaderHelper.CommentPrefix, expectedFileHeader, newLineText).Add(newLineTrivia).Add(newLineTrivia);
+
+            // Skip blank lines already at the beginning of the document, since we add our own
+            var leadingTrivia = root.GetLeadingTrivia();
+            var skipCount = 0;
+            for (var i = 0; i < leadingTrivia.Count; i++)
+            {
+                if (leadingTrivia[i].RawKind == FileHeaderHelper.EndOfLineTriviaKind)
+                {
+                    skipCount = i + 1;
+                }
+                else if (leadingTrivia[i].RawKind != FileHeaderHelper.WhitespaceTriviaKind)
+                {
+                    break;
+                }
+            }
+
+            newTrivia = newTrivia.AddRange(leadingTrivia.Skip(skipCount));
+
+            return root.WithLeadingTrivia(newTrivia);
+        }
+
+        private SyntaxTriviaList CreateNewHeader(string prefixWithLeadingSpaces, string expectedFileHeader, string newLineText)
+        {
+            var copyrightText = GetCopyrightText(prefixWithLeadingSpaces, expectedFileHeader, newLineText);
+            var newHeader = copyrightText;
+            return ParseLeadingTrivia(newHeader);
+        }
+
+        private static string GetCopyrightText(string prefixWithLeadingSpaces, string copyrightText, string newLineText)
+        {
+            copyrightText = copyrightText.Replace("\r\n", "\n");
+            var lines = copyrightText.Split('\n');
+            return string.Join(newLineText, lines.Select(line =>
+            {
+                if (string.IsNullOrEmpty(line))
+                {
+                    return prefixWithLeadingSpaces;
+                }
+                else
+                {
+                    return prefixWithLeadingSpaces + " " + line;
+                }
+            }));
+        }
+
+        private class MyCodeAction : CodeAction.DocumentChangeAction
+        {
+            public MyCodeAction(Func<CancellationToken, Task<Document>> createChangedDocument)
+                : base(FeaturesResources.Add_file_banner, createChangedDocument, nameof(AbstractFileHeaderCodeFixProvider))
+            {
+            }
+        }
+
+        private class FixAll : DocumentBasedFixAllProvider
+        {
+            private readonly AbstractFileHeaderCodeFixProvider _codeFixProvider;
+
+            public FixAll(AbstractFileHeaderCodeFixProvider codeFixProvider)
+            {
+                _codeFixProvider = codeFixProvider;
+            }
+
+            protected override string CodeActionTitle => FeaturesResources.Add_file_banner;
+
+            protected override Task<SyntaxNode> FixAllInDocumentAsync(FixAllContext fixAllContext, Document document, ImmutableArray<Diagnostic> diagnostics)
+            {
+                if (diagnostics.IsEmpty)
```

Comment:
when does this happen?

Reply:
feels super weird.

Reply:
It appears to be reachable through public APIs. Fixed the result for this case though when NRT was enabled.